src/register/pmpcfgx.rs

```diff
@@ -1,23 +1,350 @@
 /// Physical memory protection configuration
+use bit_field::BitField;
+
+/// Permission enum contains all possible permission modes for pmp registers
+#[derive(Clone, Copy, Debug)]
+pub enum Permission {
+    NONE = 0,
+    R = 1,
+    W = 2,
+    RW = 3,
+    X = 4,
+    RX = 5,
+    WX = 6,
+    RWX = 7,
+}
+
+/// Range enum contains all possible addressing modes for pmp registers
+#[derive(Clone, Copy, Debug)]
+pub enum Range {
+    OFF = 0,
+    TOR = 1,
+    NA4 = 2,
+    NAPOT = 3,
+}
+
+/// PmpByte holds the a single pmp configuration
+#[derive(Clone, Copy, Debug)]
+pub struct PmpByte {
+    pub byte: u8,
+    //permission: Option<Permission>,
```

Comment:
should these comments be removed?

Reply:
Thank you for the review @almindor!  

I tested these changes out and there were some problems.  First off, the `set_permissions()` and `set_range()` functions used `_write` which caused the existing bits in the register to be overwritten.  I changed them to `_set` and all is well.  

Next, I found that the `get_range()` method was getting the wrong two bits, 4 and 5, it should be 3 and 4.  This was fixed and tested.  

I also removed the `clear_lock()` functions because they are pointless.  I forgot that once you set the lock on a pmp configuration, you cannot clear it.  It can only be cleared by power cycle.  

Finally, regarding the comments you mentioned on the `Pmpbyte` struct.  I don't like getting the byte from the Pmpcfg<x> struct, then after getting the byte having to call the individual methods for the configurations.  I would rather have `get_byte()` become `into_config()` and return a struct with the byte, range, permissions, and locked fields.  However, this will probably force the methods to be duplicated for each module instead of the single generic definition.  What do you think?